bundles/org.openhab.binding.mycroft/src/main/java/org/openhab/binding/mycroft/internal/api/dto/MessageRecognizerLoopRecordBegin.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mycroft.internal.api.dto;
+
+import org.openhab.binding.mycroft.internal.api.MessageType;
+
+/**
+ *
+ * @author Gwendal ROULLEAU - Initial contribution
+ */
+public class MessageRecognizerLoopRecordBegin extends BaseMessage {
+
+    public Context context = new Context();
+
+    public MessageRecognizerLoopRecordBegin() {
+        this.type = MessageType.recognizer_loop__record_begin;
+    }
+
+    public static class Context {
```

Comment:
Since this is the same class as in `MessageRecognizerLoopRecordEnd`, do you think it would make sense to extract it and have it only once, maybe as a `MessageContext` class?

Reply:
Could you please explain further ? For me, it is not the same class, as the message inside is different ("recognizer_loop:record_begin" vs "recognizer_loop:record_end").
(Or maybe I didn't understand the kind of mutualisation you suggest ?)

Reply:
To me it looked like some of your "main" `Message*` classes have the same `Context` or `Data` class inside. I was thinking about to extract them into one file and reference them instead of clearing inner classes.

But since these classes are DTO classes anyway, I can also live with it as they are now.

Reply:
Oh, OK, I completely miss the point, you were talking about Data and Context.
I forgot to tell, the "data"/"context" objects I saw in the json messages exchanged between Mycroft modules are not consistent with each other, so even if they are indeed the same in this particular case (all empty), it is indeed another argument to let it that way.